packages/connect-node/conformance/known-failing-client.txt

```diff
@@ -0,0 +1,3 @@
+# When using the callback client, cancellations are calling the end-of-stream callback with an undefined error if a call is cancelled.
```

Comment:
I know this is documented as intentional, but conformance tests verify that a cancelled error is returned. Does it make sense to:

* Leave the behavior as-is and just mark these as known-failing
* Fix the behavior so that the cancelled error is passed to the `close` callback
* Modify the clients to manually create a cancelled error if the cancelFn is invoked

Reply:
We cannot change the behavior of the callback, that would be a breaking change.

The known-failing list applies to promise client and callback client, right? I'm worried that adding the known-failing for the callback client will shadow regressions in the promise client. This seems like a bad idea.

We either need a failure-list specific to callback-clients, or we have to return a synthetic ConnectError with code `canceled` to the test runner (but obviously only in the very specific case that we cancelled the call and the close callback receives no error). I would prefer a separate failure-list because the code under test should be as simple as possible.